### PR TITLE
adding pub_url method

### PR DESCRIPTION
**Pull request by pallavi-pannu-oc**
https://github.com/oneconvergence/gpuaas/issues/6988

### CODE REVIEW

File: dkube/sdk/internal/api_base.py [hunk 1]
```diff
@@ -113,7 +113,10 @@ def delete_run(self, category, user, name):
 
     def create_repo(self, repo):
         self.update_tags(repo.datum)
-        response = self._api.datums_add_one(user=repo.user, data=repo.datum)
+        ext=''
```
Comment (mak-454) [inline dkube/sdk/internal/api_base.py:116]: can you remove these lines and call as 
self._api.datums_add_one(user=repo.user, data=repo.datum, extract=str(repo.extract).lower())

File: dkube/sdk/rsrcs/model.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def __init__(self, user, name=generate("dataset"), tags=None):
         self.datum = DatumModel(name=None, tags=None, _class='model',
                                 dvs=None, source='dvs', url=None, remote=False, gitaccess=self.gitaccess,
                                 s3access=self.s3access, nfsaccess=self.nfsaccess, gcsaccess=self.gcsaccess)
-
+        self.extract = False
```
Comment (mak-454) [inline dkube/sdk/rsrcs/model.py:97]: you would need this in dataset.py too

Comment (pallavi-pannu-oc) [reply dkube/sdk/rsrcs/model.py:97]: Done

File: dkube/sdk/internal/api_base.py (hunk 1 above)
Comment (pallavi-pannu-oc) [reply dkube/sdk/internal/api_base.py]: Done.

**Pull request merged**